### PR TITLE
Move address related classes from card to ui-core

### PR DESCRIPTION
**Pull request by onurkaragunlu**
it was moved address related classes , strings and unit tests from card to ui-core

## Checklist 
- [x] Changes are tested manually

COAND-628

**Breaking Change**
Address styles have been changed. The changes are below.

from AdyenCheckout.Card.StreetInput to AdyenCheckout.StreetInput
from AdyenCheckout.Card.StreetInput.Optional to AdyenCheckout.StreetInput.Optional
from AdyenCheckout.Card.AddressInput to AdyenCheckout.AddressInput
from AdyenCheckout.Card.AddressInput.Optional to AdyenCheckout.AddressInput.Optional
from AdyenCheckout.Card.HouseNumberInput to AdyenCheckout.HouseNumberInput
from AdyenCheckout.Card.HouseNumberInput.Optional to AdyenCheckout.HouseNumberInput.Optional
from AdyenCheckout.Card.PostalCodeInput to AdyenCheckout.PostalCodeInput
from AdyenCheckout.Card.PostalCodeInput.Optional to AdyenCheckout.PostalCodeInput.Optional
from AdyenCheckout.Card.ZipCodeInput to AdyenCheckout.ZipCodeInput
from AdyenCheckout.Card.ZipCodeInput.Optional to AdyenCheckout.ZipCodeInput.Optional
from AdyenCheckout.Card.CityInput to AdyenCheckout.CityInput
from AdyenCheckout.Card.CityInput.Optional to AdyenCheckout.CityInput.Optional
from AdyenCheckout.Card.CityTownInput to AdyenCheckout.CityTownInput
from AdyenCheckout.Card.CityTownInput.Optional to AdyenCheckout.CityTownInput.Optional
from AdyenCheckout.Card.ApartmentSuiteInput to AdyenCheckout.ApartmentSuiteInput
from AdyenCheckout.Card.ApartmentSuiteInput.Optional to AdyenCheckout.ApartmentSuiteInput.Optional
from AdyenCheckout.Card.ProvinceTerritoryInput to AdyenCheckout.ProvinceTerritoryInput
from AdyenCheckout.Card.ProvinceTerritoryInput.Optional to AdyenCheckout.ProvinceTerritoryInput.Optional

### CODE REVIEW

File: ui-core/src/main/java/com/adyen/checkout/components/ui/view/address/AddressSpecification.kt [hunk 1]
```diff
@@ -3,14 +3,13 @@
  *
  * This file is open source and available under the MIT license. See the LICENSE file for more info.
  *
- * Created by ozgur on 31/10/2022.
- *
+ * Created by onurk on 13/1/2023.
  */
 
-package com.adyen.checkout.card.ui
```
Comment (OscarSpruit) [inline ui-core/src/main/java/com/adyen/checkout/components/ui/view/address/AddressSpecification.kt:10]: This class should be in `com.adyen.checkout.components.ui` as well I think

**Review by etman55 — CHANGES_REQUESTED** [review]

File: card/src/main/res/template/values/strings.xml.tt [hunk 2]
```diff
@@ -30,32 +30,6 @@
...
 
-    <string name="checkout_address_form_billing_address_title">%%billingAddress%%</string>
-    <string name="checkout_address_form_street_hint">%%street%%</string>
-    <string name="checkout_address_form_address_hint">%%address%%</string>
-    <string name="checkout_address_form_house_number_hint">%%houseNumberOrName%%</string>
-    <string name="checkout_address_form_apartment_suite_hint">%%apartmentSuite%%</string>
-    <string name="checkout_address_form_city_hint">%%city%%</string>
-    <string name="checkout_address_form_country_hint">%%country%%</string>
-    <string name="checkout_address_form_states_hint">%%state%%</string>
-    <string name="checkout_card_postal_code_hint">%%postalCode%%</string>
```
Comment (etman55) [inline card/src/main/res/template/values/strings.xml.tt:41]: I think this should be changed to `checkout_address_postal_code_hint`

File: ui-core/src/main/java/com/adyen/checkout/components/ui/AddressInputModel.kt [hunk 3]
```diff
@@ -6,7 +6,7 @@
  * Created by ozgur on 8/3/2022.
  */
 
-package com.adyen.checkout.card
+package com.adyen.checkout.components.ui
```
Comment (etman55) [inline ui-core/src/main/java/com/adyen/checkout/components/ui/AddressInputModel.kt:9]: I think this should be inside `components-core`

File: ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt [hunk 4]
```diff
@@ -6,7 +6,7 @@
  * Created by onurk on 3/1/2023.
  */
 
-package com.adyen.checkout.card
+package com.adyen.checkout.components.ui
```
Comment (etman55) [inline ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt:9]: this should be inside `components-core`

File: card/src/main/res/template/values/strings.xml.tt (hunk 2 above)
[the remark this answers is not in the document]
Comment (onurkaragunlu) [reply card/src/main/res/template/values/strings.xml.tt:41]: good catch. I missed here

[the remark this answers is not in the document]
Comment (onurkaragunlu) [reply card/src/main/res/template/values/strings.xml.tt:41]: No need to comment here. I deleted it :) Could you remove change request?

[the remark this answers is not in the document]
Comment (onurkaragunlu) [reply card/src/main/res/template/values/strings.xml.tt:41]: same here

File: ui-core/src/main/java/com/adyen/checkout/components/ui/view/address/AddressSpecification.kt (hunk 1 above)
Comment (onurkaragunlu) [reply ui-core/src/main/java/com/adyen/checkout/components/ui/view/address/AddressSpecification.kt:10]: It is under the com.adyen.checkout.components.ui.view.address. Do you mean I  exactly should put it under com.adyen.checkout.components.ui?

File: ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt (hunk 4 above)
Comment (onurkaragunlu) [reply ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt:9]: I am not sure that is the correct place. ButtonDelegate is in ui-core so AddressDelegate should be in the same place. If I move it, I have to move UI-related classes such as AddressOutputData, AddressInputModel

Comment (etman55) [reply ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt:9]: In my opinion `ButtonDelegate` is specific for a UI element which is a `Button`, but  for `AddressDelegate` it has output and input more for a complete layout not just for a single view or something also we can discuss this with rest of the team.

File: card/src/main/res/template/values/strings.xml.tt (hunk 2 above)
[the remark this answers is not in the document]
Comment (etman55) [reply card/src/main/res/template/values/strings.xml.tt:41]: yeah my bad.

**Review by etman55 — APPROVED** [review]

File: ui-core/src/main/java/com/adyen/checkout/components/ui/view/address/AddressSpecification.kt (hunk 1 above)
Comment (OscarSpruit) [reply ui-core/src/main/java/com/adyen/checkout/components/ui/view/address/AddressSpecification.kt:10]: Yes, that's what I mean

**Review by OscarSpruit — APPROVED** [review]

File: ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt (hunk 4 above)
Comment (onurkaragunlu) [reply ui-core/src/main/java/com/adyen/checkout/components/ui/AddressDelegate.kt:9]: I discussed with Joseph and Ozgur.  we decided that it should be in ui-core

**Pull request merged**